industry_benchmarks/utils/data_gathering.py

```diff
@@ -0,0 +1,511 @@
+    for edge in input_ligand_network.edges:
+
+        name = f'edge_{edge.componentA.name}_{edge.componentB.name}'
+        transformations_scores[name] = {}
+        edge_scores = {}
+        lomap_score = get_lomap_score(edge)
+        edge_scores["lomap_score"] = lomap_score
+        alchemical_charge_difference = get_alchemical_charge_difference(edge)
+        edge_scores["alchemical_charge_difference"] = alchemical_charge_difference
+        charge_score = get_charge_score(edge)
+        edge_scores["charge_score"] = charge_score
+        shape_score = get_shape_score(edge)
+        edge_scores["shape_score"] = shape_score
+        volume_score = get_volume_score(edge)
+        edge_scores["volume_score"] = volume_score
+        mapping_rmsd_score = get_mapping_RMSD_score(edge)
+        edge_scores["mapping_rmsd_score"] = mapping_rmsd_score
+        size_core, size_dummy_A, size_dummy_B = get_number_heavy_dummy_heavy_core_atoms(edge)
+        edge_scores["num_heavy_core"] = size_core
+        edge_scores["num_heavy_dummy_A"] = size_dummy_A
+        edge_scores["num_heavy_dummy_B"] = size_dummy_B
+        diff_rings_AB = get_changing_number_rings(edge)
+        diff_rot_bonds_AB = get_changing_number_rotatable_bonds(edge)
+        edge_scores["difference_num_rings_AB"] = diff_rings_AB
+        edge_scores["difference_num_rot_bonds_AB"] = diff_rot_bonds_AB
+        morgan_fp_score = get_fingerprint_similarity_score(edge)
+        edge_scores["morgan_tanimoto_similarity"] = morgan_fp_score
+        diff_sasa_AB = get_difference_solvent_accessible_surface_area(edge)
+        edge_scores["difference_solvent_accessible_surface_area"] = diff_sasa_AB
+
+        transformations_scores[name] = edge_scores
+
+    return transformations_scores
+
+
+def gather_ligand_scores(
+    input_ligand_network: LigandNetwork,
+) -> dict[str, dict[str, int]]:
+    """
+    Gather all scores for the ligands in a dict.
+    Ligand information
+        Number of rotatable bonds
+        Number of rings
+        Number of heavy atoms
+        system element counts
+    Returns
+    -------
+    dict[str, dict[str, int]]
+      Dictionary of the ligand name and a dictionary of score name and value.
+    """
+    all_ligand_scores = {}
+    for node in input_ligand_network.nodes:
+        name = f'ligand_{node.name}'
+        all_ligand_scores[name] = {}
+        ligand_scores = {}
+        num_rotatable_bonds = get_number_rotatable_bonds(node)
+        ligand_scores["num_rotatable_bonds"] = num_rotatable_bonds
+        num_rings = get_number_ring_systems(node)
+        ligand_scores["num_rings"] = num_rings
+        num_heavy_atoms = get_number_heavy_atoms(node)
+        ligand_scores["num_heavy_atoms"] = num_heavy_atoms
+        num_elements = get_system_element_count(node)
+        ligand_scores["num_elements"] = num_elements
+        sasa = get_solvent_accessible_surface_area(node)
+        ligand_scores["solvent_accessible_surface_area"] = sasa
+
```

Comment:
Lets add the formal charge of the ligand as well.

Reply:
Done in commit [1ebe9af]